pragma/publisher/perp_fetchers/binance.py

```diff
@@ -0,0 +1,124 @@
+        async with session.get(queried_url) as resp:
+            if resp.status == 404:
+                return PublisherFetchError(
+                    f"No data found for {'/'.join(pair)} from Binance"
+                )
+
+            content_type = resp.content_type
+            if content_type and "json" in content_type:
+                text = await resp.text()
+                result = json.loads(text)
+            else:
+                raise ValueError(f"Binance: Unexpected content type: {content_type}")
+
+            volume_arr = await self.fetch_volume(asset, session)
+            return self._construct(asset, result, volume_arr)
+
+    async def fetch(self, session: ClientSession):
+        entries = []
+        for asset in self.assets:
+            if asset["type"] != "FUTURE":
+                logger.debug("Skipping Binance for non-future asset %s", asset)
+                continue
+            future_entries = await self._fetch_pair(asset, session)
+            if isinstance(future_entries, list):
+                entries.extend(future_entries)
+            else:
+                entries.append(future_entries)
+        return entries
+
+    def format_url(self, quote_asset, base_asset):
+        return self.BASE_URL
+
+    def retrieve_volume(self, asset, volume_arr):  # pylint: disable=no-self-use
+        for list_asset, list_vol in volume_arr:
+            if asset == list_asset:
+                return list_vol
+        return 0
+
+    def _construct(self, asset, result, volume_arr) -> List[FutureEntry]:
+        pair = asset["pair"]
+        result_arr = []
+        for data in result:
+            timestamp = int(data["time"])
+            price = float(data["markPrice"])
+            price_int = int(price * (10 ** asset["decimals"]))
+            pair_id = currency_pair_to_pair_id(*pair)
+            volume = float(self.retrieve_volume(data["pair"], volume_arr)) / (
+                10 ** asset["decimals"]
+            )
+            if data["pair"] == f"{pair[0]}{pair[1]}":
+                expiry_timestamp = 0
+            else:
+                date_arr = data["symbol"].split("_")
+                if len(date_arr) > 1:
+                    date_part = date_arr[1]
+                    expiry_date = datetime.strptime(date_part, "%y%m%d")
+                    expiry_date = expiry_date.replace(
+                        hour=8, minute=0, second=0, tzinfo=timezone.utc
+                    )
+                    expiry_timestamp = int(expiry_date.timestamp())
+                else:
+                    expiry_timestamp = int(0)
+            result_arr.append(
+                FutureEntry(
+                    pair_id=pair_id,
+                    price=price_int,
+                    volume=int(volume),
+                    timestamp=int(timestamp / 1000),
+                    source=self.SOURCE,
+                    publisher=self.publisher,
+                    expiry_timestamp=expiry_timestamp * 1000,
```

Comment:
its a perp fetcher and still has an expiry?

Reply:
ooops, copy/paste error, not it's always zero